client/src/components/auth/Register.js

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useContext } from "react";
+import { AuthContext } from "../../providers/AuthContext";
+import Container from "@material-ui/core/Container";
+import Typography from "@material-ui/core/Typography";
+import FormGroup from "@material-ui/core/FormGroup";
+import Button from "@material-ui/core/Button";
+import FormControl from "@material-ui/core/FormControl";
+import FormHelperText from "@material-ui/core/FormHelperText";
+import InputLabel from "@material-ui/core/InputLabel";
+import OutlinedInput from "@material-ui/core/OutlinedInput";
+
+const Register = () => {
+  const auth = useContext(AuthContext);
+  const [email, setEmail] = useState(null);
+  const [password, setPassword] = useState(null);
+  const [passwordConfirmation, setPasswordConfirmation] = useState(null);
+  const [first_name, setFirst_name] = useState(null);
+  const [last_name, setLast_name] = useState(null);
+
+  const handleSubmit = e => {
+    e.preventDefault();
+
+    const { handleRegister, history } = auth;
+
+    if (password === passwordConfirmation)
+      handleRegister(
+         email, password, passwordConfirmation, first_name, last_name,
+        history
```

Comment:
We don't need to pass history in, we can use `const history = useHistory()` directly in our `useAuth` hook :D

Reply:
ohhhh yeah!

Reply:
I'm having trouble how to use history in the functions I created in AuthContext

Reply:
we can do a little chat session if your still stuck on things sometime today